lib/spack/docs/tutorial/examples/Autotools/autotools_class.py

```diff
@@ -0,0 +1,460 @@
+##############################################################################
```

Comment:
Is this just a straight copy of `lib/spack/spack/build_systems/autotools.py`? That file is guaranteed to change, meaning that this file will gradually diverge and no one will notice. Instead of copying the entire file, ask users to run:
```
$ spack edit --build-system autotools
```
Same for the other build system files.

Reply:
Ah good point. I'm hoping some of the examples I have shown don't change drastically. I only tried to display some of the relevant functions and properties needed for writing a package file.